gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -401,15 +733,230 @@ gst_tensor_src_iio_finalize (GObject * object)
   //FIXME: fill this function
 }
 
+/**
+ * @brief write the string in to the file
+ */
+static gboolean
+gst_tensor_write_sysfs_string (GstTensorSrcIIO * self, const gchar * file,
+    const gchar * base_dir, const gchar * contents)
+{
+  gchar *filename = NULL;
+  GError *error = NULL;
+  gboolean ret = FALSE;
+
+  filename = g_strdup_printf ("%s/%s", base_dir, file);
```

Comment:
Don't join paths with this.
Use Glib.Path for portability.

Reply:
Updated accordingly.